From ae8e6ca099a11e80f0b5d9fd8f3b43bf94dea6a8 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 30 Aug 2022 18:49:57 +0300
Subject: [PATCH] [SW-102459] gaudi3: use local boot info get function

So far Gaudi3 ppboot code called the boot info get function
implemented in bootrom. Due to changes in bootrom code this
function is about to be removed, so ppboot should use the local
boot info get function.

Change-Id: I906e2b6323e99deb1f513c57e36bded36e189bc3
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index eff8cae..18e9062 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -46,6 +46,7 @@
 #include <arcboot_export_dev.h>
 #include <address_space.h>
 #include <boot_desc.h>
+#include <boot_info.h>
 
 u32 hl_pll_timestamp_base_get(void)
 {
@@ -499,6 +500,5 @@
 
 int ppboot_boot_info_get(u32 type, void *data, u32 max_size, u32 *copy_size)
 {
-	/* arcboot function call */
-	return boot_info_get_no_merge(type, data, max_size, copy_size);
+	return boot_info_get(type, data, max_size, copy_size);
 }
